pyproject.toml

```diff
@@ -22,6 +22,17 @@ dependencies = [
 ]
 dynamic = ["version"]
 
+requires-python = ">=3.8"
```

Comment:
scikit-learn requires at least 3.9 (e.g [classifier](https://pypi.org/project/scikit-learn/))
I don't know to which extent skglm would work with lower version

Reply:
3.8 has reached its EOL 6 months ago so go for 3.9 !

Reply:
Seems reasonable. I originally set this to 3.11 actually, but then saw this:

`ERROR: Package 'skglm' requires a different Python: 3.8.1 not in '>=3.9'`

in the circle CI test logs, so I figured you needed 3.8. But the error still remains so I guess it's something different. 

Scipy and numpy both require at least 3.11 now, numba 3.10.

Reply:
I just bumped the version of Python used in our circleci workflow to 3.10 instead